rootfs/api/models/build.py

```diff
@@ -52,8 +52,8 @@ def version(self):
         return 'git-{}'.format(self.sha) if self.source_based else 'latest'
 
     def create(self, user, *args, **kwargs):
-        latest_release = self.app.release_set.latest()
-
+        latest_release = self.app.release_set.filter(failed=False).latest()
```

Comment:
This PR doesn't change the current flow/functionality anyway. The only thing this does is to save the build,config,release objects even if they failed instead of deleting so that a new release gets a new version number always instead of https://github.com/deis/controller/issues/989#issue-171929174. If the `deis config/build` fails you run the same command again as you do it today